config/settings/base.py

```diff
@@ -343,12 +344,34 @@ def env(name, default=NoDefaultValue, type_=str):
 GITHUB_OAUTH_SCOPES.append("repo" if GITHUB_OAUTH_PRIVATE_REPO else "public_repo")
 
 # SF client settings:
-SF_CALLBACK_URL = env("SF_CALLBACK_URL", default=None)
+SFDX_CLIENT_CALLBACK_URL = env(
+    "SFDX_CLIENT_CALLBACK_URL", default=env("SF_CALLBACK_URL", default=None)
+)
+SFDX_CLIENT_ID = env("SFDX_CLIENT_ID", default=env("SF_CLIENT_ID", default=None))
+SFDX_CLIENT_SECRET = env(
+    "SFDX_CLIENT_SECRET", default=env("SF_CLIENT_SECRET", default=None)
+)
+SFDX_SIGNUP_INSTANCE = env(
+    "SFDX_SIGNUP_INSTANCE", default=env("SF_SIGNUP_INSTANCE", default=None)
+)
 # Ugly hack to fix https://github.com/moby/moby/issues/12997
-SF_CLIENT_KEY = env("SF_CLIENT_KEY", default="").replace("\\n", "\n")
-SF_CLIENT_ID = env("SF_CLIENT_ID", default=None)
-SF_CLIENT_SECRET = env("SF_CLIENT_SECRET", default=None)
-SF_SIGNUP_INSTANCE = env("SF_SIGNUP_INSTANCE", default=None)
+DOCKER_SFDX_HUB_KEY = env("DOCKER_SFDX_HUB_KEY", default="").replace("\\n", "\n")
+SFDX_HUB_KEY = env(
+    "SFDX_HUB_KEY", default=env("SF_CLIENT_KEY", default=DOCKER_SFDX_HUB_KEY)
+)
+
+if not SFDX_CLIENT_SECRET:
+    raise ImproperlyConfigured("Missing environment variable: SFDX_CLIENT_SECRET.")
+if not SFDX_CLIENT_CALLBACK_URL:
+    raise ImproperlyConfigured(
+        "Missing environment variable: SFDX_CLIENT_CALLBACK_URL."
+    )
+if not SFDX_CLIENT_ID:
+    raise ImproperlyConfigured("Missing environment variable: SFDX_CLIENT_ID.")
```

Comment:
We should check for SFDX_HUB_KEY too. It's not required for MetaDeploy unless you're using the scratch org feature, but it is required for Metecho.